docs/2.7.0/docs/daml-finance/tutorials/instrument-extension-packages/callable-bond.rst

```diff
@@ -0,0 +1,159 @@
+define the state of the instrument. Instead, the lifecycling framework requires an active Election
+on each date when it is possible to call the bond. This Election is the main topic of this tutorial.
+Check out the :ref:`Lifecycling concepts <time-vs-election-lifecycling>` for more details on time
+based vs election based evolution of instruments.
+
+In this tutorial, we are going to:
+
+#. create a callable bond instrument and book a holding on it
+#. reuse the lifecycle rule and settlement factory from the fixed rate bond tutorial
+#. create the election not to call the bond
+#. process the election event to produce the effects of a coupon payment
+#. instruct settlement by presenting a bond holding
+#. settle the resulting batch atomically
+
+This example builds on the previous :doc:`Time-based Lifecycling <fixed-rate-bond>` tutorial script
+in the sense that the same parties, accounts, lifecycle rule and settlment factory are used.
+
+Run the Script
+**************
+
+The code for this tutorial can be executed via the ``runCallableBond`` function in the
+``CallableBond.daml`` module.
+
+Instrument and Holding
+======================
+
+In order to demonstrate the
+:ref:`Election <module-daml-finance-interface-lifecycle-election-24570>` concept, we need a suitable
+sample instrument.
+:ref:`Callable bonds <module-daml-finance-instrument-bond-callable-instrument-83330>` pay coupons as
+long as the bond has not yet been called. The :doc:`Bond Extension tutorial <bond-extension>`
+describes this instrument in more detail. Here, we briefly show how to create the bond instrument
+using a factory:
+
+.. literalinclude:: ../../finance-lifecycling/daml/Scripts/CallableBond.daml
+  :language: daml
+  :start-after: -- CREATE_CALLABLE_BOND_INSTRUMENT_BEGIN
+  :end-before: -- CREATE_CALLABLE_BOND_INSTRUMENT_END
+
+Compared to the fixed rate bond, notice that this callable instrument also has a ``callSchedule``,
+that specifies on which dates the issuer can call the bond.
+
+We also create a bond holding in Bob's account:
+
+.. literalinclude:: ../../finance-lifecycling/daml/Scripts/CallableBond.daml
+  :language: daml
+  :start-after: -- CREATE_CALLABLE_BOND_HOLDING_BEGIN
+  :end-before: -- CREATE_CALLABLE_BOND_HOLDING_END
+
+Now, we have both an instrument definition and a holding. Let us proceed to lifecycle the bond using
+``Elections``, which is the main purpose of this tutorial.
+
+Lifecycle Events and Rule
+=========================
+
+We start by creating an Election factory, which can be used to create elections:
+
+.. literalinclude:: ../../finance-lifecycling/daml/Scripts/CallableBond.daml
+  :language: daml
+  :start-after: -- CREATE_ELECTION_FACTORY_BEGIN
+  :end-before: -- CREATE_ELECTION_FACTORY_END
+
+
+An :ref:`Election <module-daml-finance-interface-lifecycle-election-24570>`
+contains two main pieces of information: the election tag (e.g. "CALLED") and the date to which it
```

Comment:
To be fair, there are three main pieces of info
- tag
- time
- who is making the election